packages/eslint-plugin/src/rules/no-confusing-void-expression.ts

```diff
@@ -388,5 +415,56 @@ export default createRule<Options, MessageId>({
       const type = getConstrainedTypeAtLocation(services, targetNode);
       return tsutils.isTypeFlagSet(type, ts.TypeFlags.VoidLike);
     }
+
+    function isFunctionReturnTypeIncludesVoid(functionType: ts.Type): boolean {
+      const callSignatures = tsutils.getCallSignaturesOfType(functionType);
+
+      return callSignatures.some(signature => {
```

Comment:
[Testing] No tests fail if this is changed to `.every`. I think that means there's some logic being missed?

Reply:
@JoshuaKGoldberg is it possible for a function expression (arrow or standard) to have multiple call signatures (other than function overloading)? I've currently changed this part of the code to assume only one call signature is available.

Reply:
Okay, I figured this out and added the necessary test. 👍